bin/email_alert_service

```diff
@@ -2,26 +2,32 @@
 
 require_relative "../email_alert_service/environment"
 rabbitmq_options = EmailAlertService.config.rabbitmq
-logger = EmailAlertService.config.logger
 
-connection = AMQPConnection.new(rabbitmq_options)
-connection.start
+amqp_options = rabbitmq_options[:amqp]
+exchange_name = rabbitmq_options[:exchange]
+queues_options = rabbitmq_options[:queues]
 
-queue_binding = MajorChangeQueue.new(connection).bind
+logger = EmailAlertService.config.logger
 
-message_processor = MessageProcessor.new(connection.channel, logger)
+connection = AMQPConnection.new(amqp_options, exchange_name)
+connection.start
 
-listener = Listener.new(queue_binding, message_processor)
+listeners = queues_options.map do |queue_options|
+  queue = EmailAlertQueue.new(connection: connection, routing_key: queue_options[:routing_key], name: queue_options[:name])
+  processor = queue_options[:processor].camelize.constantize.new(connection.channel, logger)
+  Listener.new(queue.bind, processor)
+  logger.info "Bound to exchange #{exchange_name} on queue #{queue_options[:queue_name]}"
```

Comment:
I'm a bit confused about this. This expression is in a block, that is passed to map, and it's the last expression, so `listeners` should be an array of whatever `logger.info` evaluates to?

This confuses me even more, as `listeners` is used later on... and I would guess that the return value of `logger.info` doesn't have a `listen` method.

Reply:
You're confused because this code is clearly broken.

Fixed it now.